### PR TITLE
add missing base64 encoding as error cause to message

### PR DESCRIPTION
**Pull request by schilken**
I added the possible cause "your --DartDefines are not base64 encoded" to the error message. 

This will be helpful for all developers who use something like the follwing to build their app for CI:
./gradlew app:assembleDebug -Pdart-defines=WHATEVER_FLAG=true 

*List which issues are fixed by this PR. You must list at least one issue.*
#95116

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I changed a test to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests  in assemble_test.dart are passing.

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/assemble_test.dart [hunk 1]
```diff
@@ -113,7 +113,9 @@ void main() {
     ];
     expect(
       commandRunner.run(command),
-      throwsToolExit(message: 'Error parsing assemble command: your generated configuration may be out of date')
+      throwsToolExit(
+            message:
+                'Error parsing assemble command: your --DartDefines are not base64 encoded or your generated configuration may be out of date')
```
Comment (albertodev01) [inline packages/flutter_tools/test/commands.shard/hermetic/assemble_test.dart:118]: chore: I would add a comma at the end of the line so the formatter can automatically print the parenthesis on a new line

```suggestion
                'Error parsing assemble command: your --DartDefines are not base64 encoded or your generated configuration may be out of date',)
```

Comment (albertodev01) [inline packages/flutter_tools/test/commands.shard/hermetic/assemble_test.dart:118]: chore: I would add a comma at the end of the line so the formatter can automatically print the parenthesis on a new line

```suggestion
                'Error parsing assemble command: your --DartDefines are not base64 encoded or your generated configuration may be out of date',)
```

File: packages/flutter_tools/lib/src/commands/assemble.dart [hunk 2]
```diff
@@ -295,7 +295,7 @@ class AssembleCommand extends FlutterCommand {
       decodedDefines = decodeDartDefines(environment.defines, kDartDefines);
     } on FormatException {
       throwToolExit(
-        'Error parsing assemble command: your generated configuration may be out of date. '
+        'Error parsing assemble command: your --DartDefines are not base64 encoded or your generated configuration may be out of date. '
```
Comment (albertodev01) [inline packages/flutter_tools/lib/src/commands/assemble.dart:298]: chore:

```suggestion
        'Error parsing assemble command: either your --DartDefines options aren't base64 encoded or your generated configuration may be out of date. '
```

**Review by albertodev01 — COMMENTED** [review]
I would apply these changes but other than that this looks good!

File: packages/flutter_tools/lib/src/commands/assemble.dart (hunk 2 above)
Comment (schilken) [reply packages/flutter_tools/lib/src/commands/assemble.dart]: Hi Alberto, thanks for your suggestions. I adjusted my PR accordingly with the last commit.

**Comment by christopherfujino** [comment]
@schilken I'm sorry I didn't get around to reviewing this PR earlier, and thanks for making it.

Unfortunately I disagree with @albertodev01's comments:

1. Your test change was good as is and adheres to our style guide, while breaking it onto multiple lines for the sake of the autoformatter (which we do not use in this repo) does not: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting
2. Converting `'are not'` to the contraction `"aren't"` seems unneccessary, forced you to use double quotes, whereas the style guide prefers single quotes: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-single-quotes-for-strings.

If you revert your last commit, I will happily merge this.

**Comment by skia-gold** [comment]
Gold has detected about 5 new digest(s) on patchset 5.
View them at https://flutter-gold.skia.org/cl/github/95328

**Comment by christopherfujino** [comment]
the Windows tool_integration_tests are not running because there was an upstream infrastructure change; I rebased your branch so all the tests *should* run now

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Comment by christopherfujino** [comment]
Started a thread on an internal chat room about the pending "Google testing" check.

**Comment by angjieli** [comment]
The Google3 tool have some issue with PR that has an old base commit. Can you rebase your PR @schilken ?

**Pull request closed without merging**